src/compiler/scanner.ts

```diff
@@ -1356,13 +1426,46 @@ export function createScanner(languageVersion: ScriptTarget,
         pos++;
         switch (ch) {
             case CharacterCodes._0:
-                // '\01'
-                if (isTaggedTemplate && pos < end && isDigit(text.charCodeAt(pos))) {
+                // '\08' is '\0' + '8' but is treated as an octal escape sequence, thus isDigit
+                if (!(pos < end && isDigit(text.charCodeAt(pos)))) {
+                    return "\0";
+                }
+            // '\01', '\011'
+            // falls through
+            case CharacterCodes._1:
+            case CharacterCodes._2:
+            case CharacterCodes._3:
+                // '\1', '\17', '\177'
+                if (pos < end && isOctalDigit(text.charCodeAt(pos))) {
                     pos++;
-                    tokenFlags |= TokenFlags.ContainsInvalidEscape;
-                    return text.substring(start, pos);
                 }
-                return "\0";
+            // '\17', '\177'
+            // falls through
+            case CharacterCodes._4:
+            case CharacterCodes._5:
+            case CharacterCodes._6:
+            case CharacterCodes._7:
+                // '\4', '\47' but not '\477'
+                if (pos < end && isOctalDigit(text.charCodeAt(pos))) {
+                    pos++;
+                }
+                // '\47'
+                tokenFlags |= TokenFlags.ContainsInvalidEscape;
+                if (shouldEmitInvalidEscapeError) {
+                    const code = parseInt(text.substring(start + 1, pos), 8);
+                    error(Diagnostics.Octal_escape_sequences_are_not_allowed_Use_the_syntax_0, start, pos - start, "\\x" + padLeft(code.toString(16), 2, "0"));
+                    return String.fromCharCode(code);
+                }
+                return text.substring(start, pos);
+            case CharacterCodes._8:
+            case CharacterCodes._9:
+                // the invalid '\8' and '\9'
+                tokenFlags |= TokenFlags.ContainsInvalidEscape;
+                if (shouldEmitInvalidEscapeError) {
+                    error(Diagnostics._0_is_not_allowed, start, 2, text.substring(start, pos));

```

Comment:
How about

```
Escape sequences cannot start with '{0}'
```

Reply:
Instead of hard-coding `2`, use `pos - start`.

Reply:
This is not correct, `\8` and `\9` are complete escape sequences. I've just prepended “Escape sequence” to the message.